pkg/manager/member/tidb_member_manager.go

```diff
@@ -397,15 +397,21 @@ func getTiDBConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
 		EnablePlugin:    len(plugins) > 0,
 		PluginDirectory: "/plugins",
 		PluginList:      strings.Join(plugins, ","),
+		ClusterDomain:   tc.Spec.ClusterDomain,
 	}
 
 	if tc.IsHeterogeneous() {
-		tidbStartScriptModel.Path = controller.PDMemberName(tc.Spec.Cluster.Name) + ":2379"
+		tidbStartScriptModel.Path = fmt.Sprintf("%s.%s.svc%s:2379",
```

Comment:
If tc.Spec.Cluster.Domain is not configured, we should keep the current format.

Reply:
Across Kubernetes deployment is not the same with the heterogeneous deployment, the biggest difference is that `spec.pd!=nil`.